File: frontend/refactor.js
const fs = require('fs');

let content = fs.readFileSync('App.js', 'utf8');

// 1. Add context imports
content = content.replace(
  'import React, { useState, useEffect } from "react";',
  'import React, { useState, useEffect, createContext, useContext, useCallback } from "react";\nexport const AppContext = createContext();'
);

// 2. Extract state block from DashboardScreen
const stateBlockRegex = /const \[total, setTotal\].*?const Meses = \[.*?\];/s;
const stateBlockMatch = content.match(stateBlockRegex);
if (!stateBlockMatch) throw new Error("Could not find state block");
const stateBlock = stateBlockMatch[0];

// Remove state block from DashboardScreen
content = content.replace(stateBlockRegex, '  const { total, totalLC, seletorMes, setSeletorMes, mesesDisponiveis, setVisible, visible, name, setName, valor, setValor, categoria, setCategoria, parcelas, setParcelas, grupoColuna, setGrupoColuna, handleSalvar, loadingSave, enableSave, defaultDebitMonth, defaultCreditoMonth, Meses, loading, localMessage } = useContext(AppContext);');

// 3. Extract loadTotal and handleLimpeza, etc.
// It's getting too complex. I will rewrite the file using my own logic.

File: frontend/refactor.py
import re

with open('App.js', 'r') as f:
    content = f.read()

# 1. Add createContext to React import
content = content.replace('import React, { useState, useEffect } from "react";', 'import React, { useState, useEffect, createContext, useContext } from "react";\nexport const AppContext = createContext();')

# 2. Extract state from DashboardScreen
state_block_match = re.search(r'(const \[total, setTotal\].*?const Meses = \[.*?\];)', content, re.DOTALL)
if not state_block_match:
    print("Could not find state block")
    exit(1)

state_block = state_block_match.group(1)

# We need to extract API_URL and STORAGE_KEYS as well if they are outside
# They are outside.

# 3. We will just rewrite the file by generating a new one with node or python.
# Actually, I will write the new App.js completely.
